### ISSUE TITLE
[🐛 Bug]: Constantly hanging connections and threads created by dev tools, holding connections on jdk11 http client.

### ISSUE THREAD

**Issue by pogrebnyaksa** [SeleniumHQ/selenium#11868]
### What happened?

Constantly hanging connections and threads created by dev tools, holding connections on jdk11 http client.

### How can we reproduce the issue?

```shell
public static void main(String[] args) throws InterruptedException {
        ChromeOptions chromeOptions = new ChromeOptions();
        chromeOptions.addArguments("--ignore-certificate-errors",
                "--start-maximized",
                "--no-sandbox",
                "--dns-prefetch-disable",
                "--disable-gpu");

        System.setProperty("webdriver.http.factory", "jdk-http-client");
        var start = Instant.now();
        Duration loopDuration = Duration.ofMinutes(3);
        while (Instant.now().minus(loopDuration.toMillis(), ChronoUnit.MILLIS).isBefore(start)) {
            WebDriver driver = null;
            try {
                driver = RemoteWebDriver.builder()
                        .augmentUsing(new Augmenter())
                        .oneOf(chromeOptions)
                        .address(System.getProperty("GRID_URL"))
                        .build();
                try (DevTools devTools = ((HasDevTools) driver).getDevTools()) {
                    devTools.createSession();
                    devTools.send(Network.enable(Optional.empty(), Optional.empty(), Optional.empty()));
                    devTools.send(Network.setExtraHTTPHeaders(new Headers(Map.of("test", Boolean.TRUE.toString()))));
                    devTools.addListener(Network.requestWillBeSent(), event -> {

                    });
                    driver.get("https://cnn.com");
                }
            } finally {
                if (driver != null) {
                    driver.close();
                }
            }
            TimeUnit.SECONDS.sleep(3);
        }
        while (true) {
            TimeUnit.SECONDS.sleep(10);
        }
    }

run this code for 5 mins, then using jconsole perform full GC and in thread list you can see left hanging threads even after webdriver session was closed and enough time passed to finish all threads. In production env I can see such threads hanging for days. Same issue is observable on selenium 4.8.3, devtools-v111, chromedriver 111.0.5563.64 Stacktrace of thread is:
Name: HttpClient-20-SelectorManager
State: RUNNABLE
Total blocked: 0  Total waited: 0

Stack trace: 
java.base@17.0.5/sun.nio.ch.epoll.wait(Native Method)
java.base@17.0.5/sun.nio.ch.epollselectorimpl.doselect(epollselectorimpl.java:118)
java.base@17.0.5/sun.nio.ch.selectorimpl.lockanddoselect(selectorimpl.java:129)
   - locked sun.nio.ch.Util$2@395d80e0
   - locked sun.nio.ch.EPollSelectorImpl@3fb5076d
java.base@17.0.5/sun.nio.ch.selectorimpl.select(selectorimpl.java:141)
platform/java.net.http@17.0.5/jdk.internal.net.http.httpclientimpl$selectormanager.run(httpclientimpl.java:889)

hanging connections: 
netstat -an | grep gridip
tcp        0      0 myip     gridip:4444       ESTABLISHED
tcp6       0      0 myip     gridip:4444       CLOSE_WAIT 
tcp6       1      0 myip     gridip:4444       CLOSE_WAIT 
tcp6       0      0 myip     gridip:4444       CLOSE_WAIT 
tcp6       0      0 myip     gridip:4444       CLOSE_WAIT 
tcp6       0      0 myip     gridip:4444       CLOSE_WAIT 
tcp6       0      0 myip     gridip:4444       CLOSE_WAIT 
tcp6       0      0 myip     gridip:4444       CLOSE_WAIT 
tcp6       0      0 myip     gridip:4444       CLOSE_WAIT 
tcp6       0      0 myip     gridip:4444       CLOSE_WAIT 
tcp6       1      0 myip     gridip:4444       CLOSE_WAIT 
tcp6       0      0 myip     gridip:4444       CLOSE_WAIT 
tcp6       0      0 myip     gridip:4444       CLOSE_WAIT 
tcp6       0      0 myip     gridip:4444       CLOSE_WAIT 
tcp6       0      0 myip     gridip:4444       CLOSE_WAIT 
tcp6       0      0 myip     gridip:4444       CLOSE_WAIT
```

### Relevant log output

```shell
no log output required
```

### Operating System

Debian 9, Ubuntu 20.04.3

### Selenium version

Java 4.8.0, dev tools v109

### What are the browser(s) and version(s) where you see this issue?

chrome 109.0

### What are the browser driver(s) and version(s) where you see this issue?

ChromeDriver 109.0.5414.74

### Are you using Selenium Grid?

Yes, 4.8.0

**Comment by titusfortner** [comment]
Try with driver quit instead of close.

**Comment by pogrebnyaksa** [comment]
> Try with driver quit instead of close.

With quit instead of close was the same result.
Also I should add that same issue is observable on selenium grid 4.8.3, selenium 4.8.3, devtools-v111, chromedriver 111.0.5563.64

**Comment by pogrebnyaksa** [comment]
@pujagani this issue is similar to https://github.com/SeleniumHQ/selenium/issues/11270

**Comment by joerg1985** [comment]
@pogrebnyaksa did you try the workaround with calling `System.gc` ?

**Comment by pogrebnyaksa** [comment]
> @pogrebnyaksa did you try the workaround with calling `System.gc` ?

Yes, because jconsole calls full GC and it was mentioned in issue description.

**Comment by pogrebnyaksa** [comment]
In addition, all works fine with Java 18+

**Comment by diemol** [comment]
@pogrebnyaksa can you share the updated code you are using to get this behavior? What Java version?

**Comment by pogrebnyaksa** [comment]
> @pogrebnyaksa can you share the updated code you are using to get this behavior? What Java version?

Code the same, just change Java to 18+

**Comment by pogrebnyaksa** [comment]
Despite fact that versions are not equal, it seems that https://bugs.openjdk.org/browse/JDK-8277969 relates to this task

**Comment by diemol** [comment]
And did you try with the version that has the fix?

**Comment by pogrebnyaksa** [comment]
> And did you try with the version that has the fix?

Issue reproduces on jdk17 and doesn't reproduce on jdk18 or newer(`19,20 etc.) where jdk http client was updated and probably has the fix. Jdk17 last time was updated almost 2 years ago, in 2021

**Comment by diemol** [comment]
Closed via #12035 

If it still happens after releasing 4.10.0 we can revisit this issue.
